internal/controller/controllers/local_cluster_import_controller.go

```diff
@@ -0,0 +1,711 @@
+		return err
+	}
+	return nil
+}
+
+func (r *LocalClusterImportReconciler) ImportLocalCluster() error {
+
+	var errorList error
+
+	clusterVersion, err := r.GetClusterVersion("version")
+	if err != nil {
+		r.Log.Errorf("unable to find cluster version info due to error: %s", err.Error())
+		errorList = multierror.Append(nil, err)
+		return errorList
+	}
+
+	release_image := ""
+	if clusterVersion != nil && clusterVersion.Status.History[0].State != configv1.CompletedUpdate {
+		message := "the release image in the cluster version is not ready yet, please wait for installation to complete and then restart the assisted-service"
+		r.Log.Error(message)
+		errorList = multierror.Append(errorList, errors.New(message))
+	}
+
+	kubeConfigSecret, err := r.GetSecret("openshift-kube-apiserver", "node-kubeconfigs")
+	if err != nil {
+		r.Log.Errorf("unable to fetch local cluster kubeconfigs due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	pullSecret, err := r.GetSecret("openshift-machine-api", "pull-secret")
+	if err != nil {
+		r.Log.Errorf("unable to fetch pull secret due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	dns, err := r.GetClusterDNS()
+	if err != nil {
+		r.Log.Errorf("could not fetch DNS due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	proxy, err := r.GetClusterProxy()
+	if err != nil {
+		r.Log.Errorf("could not fetch proxy due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	numberOfControlPlaneNodes, err := r.GetNumberOfControlPlaneNodes()
+	if err != nil {
+		r.Log.Errorf("unable to determine the number of control plane nodes due to error %s", err.Error())
+		errorList = multierror.Append(errorList, err)
+	}
+
+	if clusterVersion.Status.History[0].Image == "" {
+		r.Log.Error("unable to determine release image")
+		errorList = multierror.Append(errorList, err)
+	}
+
+	// If we already have errors before we start writing things, then let's stop
+	// better to let the user fix the problems than have to delete things later.
+	if errorList != nil {
+		return errorList
+	}
+
+	release_image = clusterVersion.Status.History[0].Image
+
+	err = r.createClusterImageSet(release_image)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		errorList = multierror.Append(errorList, err)
+	}
+
+	err = r.createNamespace(r.LocalClusterName)
+	if err != nil && !k8serrors.IsAlreadyExists(err) {
+		errorList = multierror.Append(errorList, err)
+	}
+
+	if kubeConfigSecret != nil {
+		err = r.createAdminKubeConfig(kubeConfigSecret)
+		if err != nil && !k8serrors.IsAlreadyExists(err) {
+			errorList = multierror.Append(errorList, err)
+		}
+	}
+
+	if pullSecret != nil {
+		err = r.createLocalClusterPullSecret(pullSecret)
+		if err != nil && !k8serrors.IsAlreadyExists(err) {
+			errorList = multierror.Append(errorList, err)
+		}
+	}
+
+	if numberOfControlPlaneNodes > 0 {
```

Comment:
How would this be 0?

Reply:
In an invalid configuration perhaps?
This simply depends on a specific label on a node. 

It could be zero if there was a bug or some error that prevented nodes from being registered?

Reply:
But we're currently running on this cluster, right?

Reply:
Yes, agreed, will remove that